### ISSUE TITLE
nx generate with directory option sets the wrong name of the project

### ISSUE THREAD

**Issue by webmandman** [nrwl/nx#6032]
## Current Behavior

This command:

`nx g library mystock --directory=data/`

generates a project name in angular.json as "data-mystock"

## Expected Behavior

I expect this to be a project named " mystock", and the --directory option is only for folder structure.

**Comment by webmandman** [comment]
I take it back. sometimes even when I only specify one folder for the --directory option it still goes through installing NPM packages.

**Comment by AgentEnder** [comment]
Hey, this is actually working as intended. Doing this prevents common naming collisions a user might experience. If the directory structure is feature based, the names tend to make more sense (e.g. mystock/data -> mystock-data). If they do not make sense for your use case, you can change the project name by changing its key in workspace/angular.json as well as nx.json

**Comment by webmandman** [comment]
ok. I specially didn't like it when I started to add components to the lib `nx g c list --project=data-stock-mystock`  I expected to just specify `--project=mystock` "mystock" being the name of the project that I chose.  I know this project name is not that long "data-stock-mystock", but I imagine longer names. 

feature request:  add an option ie: --uniquename=false that uses the --name value and not name plus directory. --uniquename is true by default.
